kernelci/build.py

```diff
@@ -1385,6 +1386,60 @@ def install(self, verbose=False):
         return super().install(verbose)
 
 
+class MakePerfTool(Step):
+
+    @property
+    def name(self):
+        return 'perf'
+
+    def is_enabled(self):
+        """Check whether the perf is enabled
+
+        Return True if perf config fragment is enabled in the build
+        meta-data and CONFIG_PERF_EVENTS is in kernel config, or False
+        otherwise.
+        """
+        return 'perf' in self._meta.get(
+            'bmeta', 'kernel', 'defconfig_extras'
+        ) and self._kernel_config_enabled('PERF_EVENTS')
+
+    def run(self, jopt=None, verbose=False, opts=None):
+        """Make the perf tool
+
+        Make the kernel perf tool and produce a binary.
+
+        *jopt* is the `make -j` option which will default to `nproc + 2`
+        *verbose* is whether the build output should be shown
+        """
+        # requires output to be in tools/perf so overwrite it here.
+        opts = {
+            'O': '',
+        }
+        res = self._make('perf', jopt, verbose, opts, 'tools/perf')
+        return self._add_run_step(res, jopt)
+
+    def install(self, verbose=False):
+        """Install the perf binary
+
+        Install the perf binary
+
+        *verbose* is whether the build output should be shown
+        """
+        perf = os.path.join(self._kdir, 'tools/perf/perf')
```

Comment:
If the only artifact is this `perf` binary, then we don't need to create a tarball.